core/modes/modes.go

```diff
@@ -77,7 +90,7 @@ func ReconstructRequest(pair models.RequestResponsePair) (*http.Request, error)
 }
 
 // ReconstructResponse changes original response with details provided in Constructor Payload.Response
-func ReconstructResponse(request *http.Request, pair models.RequestResponsePair) *http.Response {
+func ReconstructResponse(request *http.Request, pair models.RequestResponsePair) ProcessResult {
```

Comment:
can we not change the interface for this function? it's for reconstructing the HTTP response. It doesn't do much about the delay. You can create the `ProcessResult` within the `process` function, it can reduce the amount of changes in this PR.

Reply:
fixed